src/transports/dash/segment_parser.ts

```diff
@@ -26,64 +32,121 @@ import {
+                                 init,
+                                 scheduleRequest } : ISegmentParserArguments<Uint8Array |
+                                                                             ArrayBuffer |
+                                                                             null >
+) : Observable<ISegmentParserResponseEvent<Uint8Array|ArrayBuffer> |
+               ITransportRetryEvent> {
   const { period, representation, segment } = content;
   const { data, isChunked } = response;
   if (data == null) {
-    return observableOf({ chunkData: null,
-                          chunkInfos: null,
-                          chunkOffset: 0,
-                          appendWindow: [period.start, period.end] });
+    return observableOf({ type: "parser-response" as const,
+                          value: { chunkData: null,
+                                   chunkInfos: null,
+                                   chunkOffset: 0,
+                                   appendWindow: [period.start, period.end] } });
   }
-
   const chunkData = data instanceof Uint8Array ? data :
                                                  new Uint8Array(data);
 
-  const indexRange = segment.indexRange;
+  const { range } = segment;
   const isWEBM = isWEBMEmbeddedTrack(representation);
 
-  const nextSegments = isWEBM ?
-    getSegmentsFromCues(chunkData, 0) :
-    getSegmentsFromSidx(chunkData, Array.isArray(indexRange) ? indexRange[0] :
-                                                               0);
+  let nextSegments;
+  let indexes;
+
+  if (isWEBM) {
+    nextSegments = getSegmentsFromCues(chunkData, 0);
+  } else {
+    const indexOffset = Array.isArray(range) ? range[0] :
+                                               0;
+    const referencesFromSidx = getReferencesFromSidx(chunkData, indexOffset);
+    if (referencesFromSidx !== null) {
+      const [indexReferences, segmentReferences] = referencesFromSidx;
+      if (segmentReferences.length > 0) {
+        nextSegments = segmentReferences;
+      }
+      indexes = indexReferences;
+    }
+  }
 
   if (!segment.isInit) {
-    const chunkInfos = isWEBM ? null : // TODO extract from webm
-                                getISOBMFFTimingInfos(chunkData,
-                                                      isChunked,
-                                                      segment,
-                                                      init);
-    return observableOf({ chunkData,
-                          chunkInfos,
-                          chunkOffset: takeFirstSet<number>(segment.timestampOffset,
-                                                            0),
-                          appendWindow: [period.start, period.end] });
-  } else { // it is an initialization segment
-    if (nextSegments !== null && nextSegments.length > 0) {
-      representation.index._addSegments(nextSegments);
+    const initChunkInfos = isWEBM ? null : // TODO extract from webm
```

Comment:
Why would a variable for a non-init segment begin by `init`?

Reply:
Done